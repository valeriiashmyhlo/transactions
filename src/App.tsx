import React, { useState, FC } from "react";
import styles from "./App.module.css";
import { useTransactions } from "./hooks";
import { Transaction } from "./types";

type Quantity = 5 | 10 | 15 | "all";

const App: FC<{ getTransactions: () => Promise<Transaction[]> }> = ({
  getTransactions,
}) => {
  const [isFilterActive, setFilterActive] = useState(false);
  const [quantity, setQuantity] = useState<Quantity>("all");
  const [transactions, isLoaded, error] = useTransactions(getTransactions);

  const quantityOptions: Quantity[] = [5, 10, 15, "all"];

  let currentTransactions = transactions;
  if (isFilterActive) {
    currentTransactions = [...currentTransactions]
      .sort((a, b) => b.amount.value - a.amount.value)
      .filter(({ amount }) => amount.value < 0);
  }
  if (quantity !== "all") {
    currentTransactions = currentTransactions.slice(0, quantity);
  }

  return (
    <div className={styles.app}>
      <div className={styles.header}>
        <div>
          <input
            id="filter"
            type="checkbox"
            checked={isFilterActive}
            onChange={() => setFilterActive(!isFilterActive)}
          />
          <label htmlFor="filter">Smallest expenses</label>
        </div>
        <div>
          <label htmlFor="quantity">Quantity: </label>
          <select
            value={quantity}
            onChange={(e) => setQuantity(e.target.value as Quantity)}
            id="quantity"
          >
            {quantityOptions.map((item) => (
              <option key={item} value={item}>
                {item}
              </option>
            ))}
          </select>
        </div>
      </div>
      <table className={styles.table}>
        <thead>
          <tr>
            <th>Description</th>
            <th>Cost</th>
            <th>Category</th>
            <th>Date</th>
          </tr>
        </thead>
        <tbody>
          {isLoaded ? (
            error ? (
              <tr>
                <td colSpan={3}>
                  Sorry, there was an error loading your products
                </td>
              </tr>
            ) : (
              currentTransactions.map(
                ({ id, amount, description, category_title, date }) => (
                  <tr key={id}>
                    <td>{description}</td>
                    <td>
                      {amount.value} {amount.currency_iso}
                    </td>
                    <td>{category_title}</td>
                    <td>{date}</td>
                  </tr>
                )
              )
            )
          ) : (
            <tr>
              <td colSpan={3}>Loading...</td>
            </tr>
          )}
        </tbody>
      </table>
    </div>
  );
};

export default App;
