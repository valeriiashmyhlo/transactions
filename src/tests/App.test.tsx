import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import { MockTransactions } from "./mockData";
import App from "../App";

const getTransactions = async () => MockTransactions;

describe("App", () => {
  it("should render App component", async () => {
    const { container, getByText } = render(
      <App getTransactions={getTransactions} />
    );
    expect(container.firstChild).toMatchSnapshot();
    await waitFor(() => getByText("Smallest expenses"));
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should filter by smallest expenses on check "Smallest expenses" checkbox', async () => {
    const { container, getByLabelText } = render(
      <App getTransactions={getTransactions} />
    );
    const checkbox = await waitFor(() => getByLabelText("Smallest expenses"));
    fireEvent.click(checkbox);
    expect(container.firstChild).toMatchSnapshot();
    fireEvent.click(checkbox);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should filter quantity", async () => {
    const { container, getByLabelText } = render(
      <App getTransactions={getTransactions} />
    );
    const select = await waitFor(() => getByLabelText("Quantity:"));
    fireEvent.change(select, { target: { value: 5 } });
    expect(container.firstChild).toMatchSnapshot();
    fireEvent.change(select, { target: { value: 10 } });
    expect(container.firstChild).toMatchSnapshot();
    fireEvent.change(select, { target: { value: 15 } });
    expect(container.firstChild).toMatchSnapshot();
    fireEvent.change(select, { target: { value: "all" } });
    expect(container.firstChild).toMatchSnapshot();
  });
});
