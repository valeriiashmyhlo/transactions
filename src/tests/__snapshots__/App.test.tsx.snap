// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should filter by smallest expenses on check Smallest expenses checkbox 1`] = `
<div
  class="app"
>
  <div
    class="header"
  >
    <div>
      <input
        id="filter"
        type="checkbox"
      />
      <label
        for="filter"
      >
        Smallest expenses
      </label>
    </div>
    <div>
      <label
        for="quantity"
      >
        Quantity: 
      </label>
      <select
        id="quantity"
      >
        <option
          value="5"
        >
          5
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="15"
        >
          15
        </option>
        <option
          value="all"
        >
          all
        </option>
      </select>
    </div>
  </div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Cost
        </th>
        <th>
          Category
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Transport for London
        </td>
        <td>
          -2.85
           
          GBP
        </td>
        <td>
          Transport
        </td>
        <td>
          2018-07-05
        </td>
      </tr>
      <tr>
        <td>
          Costa Coffee
        </td>
        <td>
          -3.75
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-04
        </td>
      </tr>
      <tr>
        <td>
          Wilko
        </td>
        <td>
          -10.76
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Pact Coffee
        </td>
        <td>
          -19.99
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Vodafone
        </td>
        <td>
          -29.99
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Tesco
        </td>
        <td>
          -34.98
           
          GBP
        </td>
        <td>
          Groceries
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Pret a Manger
        </td>
        <td>
          -35.87
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-09
        </td>
      </tr>
      <tr>
        <td>
          Marks & Spencer
        </td>
        <td>
          -45.23
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          British Gas
        </td>
        <td>
          -48.75
           
          GBP
        </td>
        <td>
          Bills
        </td>
        <td>
          2018-06-29
        </td>
      </tr>
      <tr>
        <td>
          Asda
        </td>
        <td>
          -51.19
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Tesco
        </td>
        <td>
          -57.21
           
          GBP
        </td>
        <td>
          Groceries
        </td>
        <td>
          2018-06-30
        </td>
      </tr>
      <tr>
        <td>
          BrewDog Shoreditch
        </td>
        <td>
          -76.06
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-06-25
        </td>
      </tr>
      <tr>
        <td>
          John Lewis
        </td>
        <td>
          -76.45
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Vue Cinema
        </td>
        <td>
          -99.89
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -99.95
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Ebay
        </td>
        <td>
          -120.29
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -145.25
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-06-24
        </td>
      </tr>
      <tr>
        <td>
          Gaucho
        </td>
        <td>
          -151.25
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -1678.49
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-06-11
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`App should filter by smallest expenses on check Smallest expenses checkbox 2`] = `
<div
  class="app"
>
  <div
    class="header"
  >
    <div>
      <input
        id="filter"
        type="checkbox"
      />
      <label
        for="filter"
      >
        Smallest expenses
      </label>
    </div>
    <div>
      <label
        for="quantity"
      >
        Quantity: 
      </label>
      <select
        id="quantity"
      >
        <option
          value="5"
        >
          5
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="15"
        >
          15
        </option>
        <option
          value="all"
        >
          all
        </option>
      </select>
    </div>
  </div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Cost
        </th>
        <th>
          Category
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Tesco
        </td>
        <td>
          -57.21
           
          GBP
        </td>
        <td>
          Groceries
        </td>
        <td>
          2018-06-30
        </td>
      </tr>
      <tr>
        <td>
          Max Mustermann
        </td>
        <td>
          510.55
           
          GBP
        </td>
        <td>
          Income
        </td>
        <td>
          2018-06-22
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -99.95
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Transport for London
        </td>
        <td>
          -2.85
           
          GBP
        </td>
        <td>
          Transport
        </td>
        <td>
          2018-07-05
        </td>
      </tr>
      <tr>
        <td>
          Pret a Manger
        </td>
        <td>
          -35.87
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-09
        </td>
      </tr>
      <tr>
        <td>
          Vodafone
        </td>
        <td>
          -29.99
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Marks & Spencer
        </td>
        <td>
          -45.23
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          British Gas
        </td>
        <td>
          -48.75
           
          GBP
        </td>
        <td>
          Bills
        </td>
        <td>
          2018-06-29
        </td>
      </tr>
      <tr>
        <td>
          BrewDog Shoreditch
        </td>
        <td>
          -76.06
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-06-25
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -1678.49
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-06-11
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -145.25
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-06-24
        </td>
      </tr>
      <tr>
        <td>
          Wilko
        </td>
        <td>
          -10.76
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Pact Coffee
        </td>
        <td>
          -19.99
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Tesco
        </td>
        <td>
          -34.98
           
          GBP
        </td>
        <td>
          Groceries
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          John Lewis
        </td>
        <td>
          -76.45
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Asda
        </td>
        <td>
          -51.19
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Gaucho
        </td>
        <td>
          -151.25
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Vue Cinema
        </td>
        <td>
          -99.89
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Costa Coffee
        </td>
        <td>
          -3.75
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-04
        </td>
      </tr>
      <tr>
        <td>
          Ebay
        </td>
        <td>
          -120.29
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`App should filter quantity 1`] = `
<div
  class="app"
>
  <div
    class="header"
  >
    <div>
      <input
        id="filter"
        type="checkbox"
      />
      <label
        for="filter"
      >
        Smallest expenses
      </label>
    </div>
    <div>
      <label
        for="quantity"
      >
        Quantity: 
      </label>
      <select
        id="quantity"
      >
        <option
          value="5"
        >
          5
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="15"
        >
          15
        </option>
        <option
          value="all"
        >
          all
        </option>
      </select>
    </div>
  </div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Cost
        </th>
        <th>
          Category
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Tesco
        </td>
        <td>
          -57.21
           
          GBP
        </td>
        <td>
          Groceries
        </td>
        <td>
          2018-06-30
        </td>
      </tr>
      <tr>
        <td>
          Max Mustermann
        </td>
        <td>
          510.55
           
          GBP
        </td>
        <td>
          Income
        </td>
        <td>
          2018-06-22
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -99.95
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Transport for London
        </td>
        <td>
          -2.85
           
          GBP
        </td>
        <td>
          Transport
        </td>
        <td>
          2018-07-05
        </td>
      </tr>
      <tr>
        <td>
          Pret a Manger
        </td>
        <td>
          -35.87
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-09
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`App should filter quantity 2`] = `
<div
  class="app"
>
  <div
    class="header"
  >
    <div>
      <input
        id="filter"
        type="checkbox"
      />
      <label
        for="filter"
      >
        Smallest expenses
      </label>
    </div>
    <div>
      <label
        for="quantity"
      >
        Quantity: 
      </label>
      <select
        id="quantity"
      >
        <option
          value="5"
        >
          5
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="15"
        >
          15
        </option>
        <option
          value="all"
        >
          all
        </option>
      </select>
    </div>
  </div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Cost
        </th>
        <th>
          Category
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Tesco
        </td>
        <td>
          -57.21
           
          GBP
        </td>
        <td>
          Groceries
        </td>
        <td>
          2018-06-30
        </td>
      </tr>
      <tr>
        <td>
          Max Mustermann
        </td>
        <td>
          510.55
           
          GBP
        </td>
        <td>
          Income
        </td>
        <td>
          2018-06-22
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -99.95
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Transport for London
        </td>
        <td>
          -2.85
           
          GBP
        </td>
        <td>
          Transport
        </td>
        <td>
          2018-07-05
        </td>
      </tr>
      <tr>
        <td>
          Pret a Manger
        </td>
        <td>
          -35.87
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-09
        </td>
      </tr>
      <tr>
        <td>
          Vodafone
        </td>
        <td>
          -29.99
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Marks & Spencer
        </td>
        <td>
          -45.23
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          British Gas
        </td>
        <td>
          -48.75
           
          GBP
        </td>
        <td>
          Bills
        </td>
        <td>
          2018-06-29
        </td>
      </tr>
      <tr>
        <td>
          BrewDog Shoreditch
        </td>
        <td>
          -76.06
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-06-25
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -1678.49
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-06-11
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`App should filter quantity 3`] = `
<div
  class="app"
>
  <div
    class="header"
  >
    <div>
      <input
        id="filter"
        type="checkbox"
      />
      <label
        for="filter"
      >
        Smallest expenses
      </label>
    </div>
    <div>
      <label
        for="quantity"
      >
        Quantity: 
      </label>
      <select
        id="quantity"
      >
        <option
          value="5"
        >
          5
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="15"
        >
          15
        </option>
        <option
          value="all"
        >
          all
        </option>
      </select>
    </div>
  </div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Cost
        </th>
        <th>
          Category
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Tesco
        </td>
        <td>
          -57.21
           
          GBP
        </td>
        <td>
          Groceries
        </td>
        <td>
          2018-06-30
        </td>
      </tr>
      <tr>
        <td>
          Max Mustermann
        </td>
        <td>
          510.55
           
          GBP
        </td>
        <td>
          Income
        </td>
        <td>
          2018-06-22
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -99.95
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Transport for London
        </td>
        <td>
          -2.85
           
          GBP
        </td>
        <td>
          Transport
        </td>
        <td>
          2018-07-05
        </td>
      </tr>
      <tr>
        <td>
          Pret a Manger
        </td>
        <td>
          -35.87
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-09
        </td>
      </tr>
      <tr>
        <td>
          Vodafone
        </td>
        <td>
          -29.99
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Marks & Spencer
        </td>
        <td>
          -45.23
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          British Gas
        </td>
        <td>
          -48.75
           
          GBP
        </td>
        <td>
          Bills
        </td>
        <td>
          2018-06-29
        </td>
      </tr>
      <tr>
        <td>
          BrewDog Shoreditch
        </td>
        <td>
          -76.06
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-06-25
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -1678.49
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-06-11
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -145.25
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-06-24
        </td>
      </tr>
      <tr>
        <td>
          Wilko
        </td>
        <td>
          -10.76
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Pact Coffee
        </td>
        <td>
          -19.99
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Tesco
        </td>
        <td>
          -34.98
           
          GBP
        </td>
        <td>
          Groceries
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          John Lewis
        </td>
        <td>
          -76.45
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`App should filter quantity 4`] = `
<div
  class="app"
>
  <div
    class="header"
  >
    <div>
      <input
        id="filter"
        type="checkbox"
      />
      <label
        for="filter"
      >
        Smallest expenses
      </label>
    </div>
    <div>
      <label
        for="quantity"
      >
        Quantity: 
      </label>
      <select
        id="quantity"
      >
        <option
          value="5"
        >
          5
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="15"
        >
          15
        </option>
        <option
          value="all"
        >
          all
        </option>
      </select>
    </div>
  </div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Cost
        </th>
        <th>
          Category
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Tesco
        </td>
        <td>
          -57.21
           
          GBP
        </td>
        <td>
          Groceries
        </td>
        <td>
          2018-06-30
        </td>
      </tr>
      <tr>
        <td>
          Max Mustermann
        </td>
        <td>
          510.55
           
          GBP
        </td>
        <td>
          Income
        </td>
        <td>
          2018-06-22
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -99.95
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Transport for London
        </td>
        <td>
          -2.85
           
          GBP
        </td>
        <td>
          Transport
        </td>
        <td>
          2018-07-05
        </td>
      </tr>
      <tr>
        <td>
          Pret a Manger
        </td>
        <td>
          -35.87
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-09
        </td>
      </tr>
      <tr>
        <td>
          Vodafone
        </td>
        <td>
          -29.99
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Marks & Spencer
        </td>
        <td>
          -45.23
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          British Gas
        </td>
        <td>
          -48.75
           
          GBP
        </td>
        <td>
          Bills
        </td>
        <td>
          2018-06-29
        </td>
      </tr>
      <tr>
        <td>
          BrewDog Shoreditch
        </td>
        <td>
          -76.06
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-06-25
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -1678.49
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-06-11
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -145.25
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-06-24
        </td>
      </tr>
      <tr>
        <td>
          Wilko
        </td>
        <td>
          -10.76
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Pact Coffee
        </td>
        <td>
          -19.99
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Tesco
        </td>
        <td>
          -34.98
           
          GBP
        </td>
        <td>
          Groceries
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          John Lewis
        </td>
        <td>
          -76.45
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Asda
        </td>
        <td>
          -51.19
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Gaucho
        </td>
        <td>
          -151.25
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Vue Cinema
        </td>
        <td>
          -99.89
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Costa Coffee
        </td>
        <td>
          -3.75
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-04
        </td>
      </tr>
      <tr>
        <td>
          Ebay
        </td>
        <td>
          -120.29
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`App should render App component 1`] = `
<div
  class="app"
>
  <div
    class="header"
  >
    <div>
      <input
        id="filter"
        type="checkbox"
      />
      <label
        for="filter"
      >
        Smallest expenses
      </label>
    </div>
    <div>
      <label
        for="quantity"
      >
        Quantity: 
      </label>
      <select
        id="quantity"
      >
        <option
          value="5"
        >
          5
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="15"
        >
          15
        </option>
        <option
          value="all"
        >
          all
        </option>
      </select>
    </div>
  </div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Cost
        </th>
        <th>
          Category
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colspan="3"
        >
          Loading...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`App should render App component 2`] = `
<div
  class="app"
>
  <div
    class="header"
  >
    <div>
      <input
        id="filter"
        type="checkbox"
      />
      <label
        for="filter"
      >
        Smallest expenses
      </label>
    </div>
    <div>
      <label
        for="quantity"
      >
        Quantity: 
      </label>
      <select
        id="quantity"
      >
        <option
          value="5"
        >
          5
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="15"
        >
          15
        </option>
        <option
          value="all"
        >
          all
        </option>
      </select>
    </div>
  </div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Cost
        </th>
        <th>
          Category
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Tesco
        </td>
        <td>
          -57.21
           
          GBP
        </td>
        <td>
          Groceries
        </td>
        <td>
          2018-06-30
        </td>
      </tr>
      <tr>
        <td>
          Max Mustermann
        </td>
        <td>
          510.55
           
          GBP
        </td>
        <td>
          Income
        </td>
        <td>
          2018-06-22
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -99.95
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Transport for London
        </td>
        <td>
          -2.85
           
          GBP
        </td>
        <td>
          Transport
        </td>
        <td>
          2018-07-05
        </td>
      </tr>
      <tr>
        <td>
          Pret a Manger
        </td>
        <td>
          -35.87
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-09
        </td>
      </tr>
      <tr>
        <td>
          Vodafone
        </td>
        <td>
          -29.99
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Marks & Spencer
        </td>
        <td>
          -45.23
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          British Gas
        </td>
        <td>
          -48.75
           
          GBP
        </td>
        <td>
          Bills
        </td>
        <td>
          2018-06-29
        </td>
      </tr>
      <tr>
        <td>
          BrewDog Shoreditch
        </td>
        <td>
          -76.06
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-06-25
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -1678.49
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-06-11
        </td>
      </tr>
      <tr>
        <td>
          Amazon
        </td>
        <td>
          -145.25
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-06-24
        </td>
      </tr>
      <tr>
        <td>
          Wilko
        </td>
        <td>
          -10.76
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Pact Coffee
        </td>
        <td>
          -19.99
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          Tesco
        </td>
        <td>
          -34.98
           
          GBP
        </td>
        <td>
          Groceries
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
      <tr>
        <td>
          John Lewis
        </td>
        <td>
          -76.45
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Asda
        </td>
        <td>
          -51.19
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Gaucho
        </td>
        <td>
          -151.25
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-02
        </td>
      </tr>
      <tr>
        <td>
          Vue Cinema
        </td>
        <td>
          -99.89
           
          GBP
        </td>
        <td>
          Entertainment
        </td>
        <td>
          2018-07-01
        </td>
      </tr>
      <tr>
        <td>
          Costa Coffee
        </td>
        <td>
          -3.75
           
          GBP
        </td>
        <td>
          Eating Out
        </td>
        <td>
          2018-07-04
        </td>
      </tr>
      <tr>
        <td>
          Ebay
        </td>
        <td>
          -120.29
           
          GBP
        </td>
        <td>
          Shopping
        </td>
        <td>
          2018-07-03
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
